gateway/ln-gateway/src/federation_manager.rs

```diff
@@ -81,77 +69,62 @@ impl FederationManager {
         }
 
         self.scid_to_federation
-            .write()
-            .await
             .retain(|_, fid| *fid != federation_id);
         Ok(())
     }
 
-    pub async fn get_client_for_scid(
-        &self,
-        short_channel_id: u64,
-    ) -> Option<Spanned<ClientHandleArc>> {
-        let scid_to_feds = self.scid_to_federation.read().await;
-        let clients = self.clients.read().await;
-
-        let federation_id = scid_to_feds.get(&short_channel_id)?;
+    pub fn get_client_for_scid(&self, short_channel_id: u64) -> Option<Spanned<ClientHandleArc>> {
+        let federation_id = self.scid_to_federation.get(&short_channel_id)?;
         // TODO(tvolk131): Cloning the client here could cause issues with client
         // shutdown (see `remove_client` above). Perhaps this function should take a
         // lambda and pass it into `client.with_sync`.
-        if let Some(client) = clients.get(federation_id).cloned() {
+        if let Some(client) = self.clients.get(federation_id).cloned() {
             Some(client)
         } else {
             error!("`FederationManager.scid_to_federation` is out of sync with `FederationManager.clients`! This is a bug.");
             None
         }
     }
 
-    pub async fn get_scid_for_federation(&self, federation_id: FederationId) -> Option<u64> {
-        self.scid_to_federation
-            .read()
-            .await
-            .iter()
-            .find_map(|(scid, fid)| {
-                if *fid == federation_id {
-                    Some(*scid)
-                } else {
-                    None
-                }
-            })
+    pub fn get_scid_for_federation(&self, federation_id: FederationId) -> Option<u64> {
+        self.scid_to_federation.iter().find_map(|(scid, fid)| {
+            if *fid == federation_id {
+                Some(*scid)
+            } else {
+                None
+            }
+        })
     }
 
-    pub async fn clone_scid_map(&self) -> BTreeMap<u64, FederationId> {
-        self.scid_to_federation.read().await.clone()
+    pub fn clone_scid_map(&self) -> BTreeMap<u64, FederationId> {
+        self.scid_to_federation.clone()
     }
 
-    pub async fn has_federation(&self, federation_id: FederationId) -> bool {
-        self.clients.read().await.contains_key(&federation_id)
+    pub fn has_federation(&self, federation_id: FederationId) -> bool {
+        self.clients.contains_key(&federation_id)
     }
 
-    // TODO(tvolk131): Sharing read lock guards could lead to write contention since
-    // we don't know how long callers will hold the guard for. Replace this function
-    // with more granular accessors that don't share lock guards.
-    pub async fn borrow_clients(
+    // TODO(tvolk131): Replace this function with more granular accessors.
+    pub fn borrow_clients(
```

Comment:
nit: just call it `clients()`?

Reply:
Kind of leaks the encapsulation. Why is this necessary currently?